### ISSUE TITLE
[DDP][Master Weight] For DDP + Master weight, is it necessary to set torch seed for training?

### ISSUE THREAD

**Issue by zejun-chen** [NVIDIA/apex#1658]
Hi,

When we are using APEX for DDP model training, is it necessary to **set the seed** at the beginning to make sure the master weight are same value among all ranks ? The PyTorch DDP broadcast only does collective operation on model's weight to keep the model are identical for all ranks, while for master weight, how does APEX to keep they are identical among all ranks?

Thank you.

**Comment by zejun-chen** [comment]
Hello, is there someone help to give a explanation ?

**Comment by crcrpar** [comment]
why would you want to use apex's DDP at this moment?
It's going to be removed in a few months

**Comment by zejun-chen** [comment]
> why would you want to use apex's DDP at this moment? It's going to be removed in a few months

Thank you for your attention.
I don't use APEX DDP, i am using torch native DDP and APEX for model training.
I want to know is it necessary to set the torch seed at the beginning when running DDP with APEX master weight ?
How does APEX make sure master weight is same on all ranks ?
Thank you.
